controllers/basic_controller.go

```diff
@@ -486,6 +488,35 @@ func (i *instanceReconcilerHelper) createOrUpdateInstance(ctx context.Context, o
 		"annotations", o.GetAnnotations(),
 	)
 
+	// We don't really know if an instance is created or updated because:
+	// 1. The go client's retry mechanism may create an instance successfully but receive a 5xx error,
+	//    causing the next attempt to get a conflict error
+	// 2. Remote state may be temporarily inconsistent, so GET checks can return false positives/negatives
+	// 3. Some handlers implement their own retry logic and keep trying to create instances until success
+	// Therefore, we can't rely on "exists" checks to determine the operation type
+	const reason = "CreateOrUpdate"
```

Comment:
nit: Conditions looks to me like events, meaning that maybe it makes sense to name then in past tense as well